pulp_2to3_migrate/app/serializers.py

```diff
@@ -29,12 +35,44 @@ def validate(self, data):
         """
         Validate that the Serializer contains valid data.
 
-        TODO:
-        Validate JSON structure of migration_plan.
-        Check validity of the JSON content:
-         - migration for requested plugins is supported
-
+        Validates JSON structure of migration_plan.
+        Checks pulp2 and pulp3 plugins are installed.
         """
+        schema = json.loads(SCHEMA)
+        validator = Draft7Validator(schema)
+        loaded_plan = json.loads(data['plan'])
+        err = []
+        for error in sorted(validator.iter_errors(loaded_plan), key=str):
+            err.append(error.message)
+        if err:
+            raise serializers.ValidationError(
+                    _("Provided Migration Plan format is invalid:'{}'".format(err))
+            )
+        plugins_to_migrate = set()
+        for plugin_type in loaded_plan['plugins']:
+            plugins_to_migrate.add(plugin_type['type'])
+        if len(loaded_plan['plugins']) != len(plugins_to_migrate):
+            raise serializers.ValidationError(
+                _("Provided Migration Plan contains same plugin type specified more that once.")
```

Comment:
i can provide specifically which plugin was specified more that once if that will be valuable info, however i will need to count the occurrences and put this into a dict

Reply:
I think it's fine as it is.
s/that/than/